### DISCUSSION TITLE
Hitting Esc key before results screen shows up will skip it

### DISCUSSION THREAD

**Discussion by respektive** [ppy/osu#17955] [Q&A] [answered]
As has been reported before: https://github.com/ppy/osu/issues/3060 https://github.com/ppy/osu/issues/13644

I believe the behaviour from stable should come back for this as I see a lot of players do this all the time and I don't think we can expect people to break this habit that easily. Even after playing more and more lazer over the last months, it is very hard for me to get rid of this habit and I still find myself hitting the escape key after almost every play. 

Maybe this isn't affecting as many players as I think though, as it's not that much of a problem with the "hold to confirm" option, which i personally have disabled. I'm interested to hear more opinions on this.

**Comment by frenzibyte** [comment]
I personally find it backwards to make <kbd>ESC</kbd> progress forward to results screen. What you are expected to press instead is the <kbd>Space</kbd> key (binding of `SkipCutscene`), and you can remap to any button you would think of as a "forward" key.

**Comment by peppy** [comment, marked as answer]
The alternative would be to block the escape press completely for that very short period of time. Or make it show the pause screen.

**Reply by respektive** [reply]
Sounds like a better way to handle this, instead of it completely skipping the results screen.

**Reply by peppy** [reply]
I'm curious which of the two proposals is more popular (block completely or allow pausing). I think the second makes more sense in terms of UX?

**Reply by respektive** [reply]
Pausing definitely makes more sense to me. Blocking input completely could cause confusion, because it isn't very clear when exactly it's being blocked. You're basically still in gameplay until the results screen shows up, so I think being able to pause might even be more expected?

**Reply by peppy** [reply]
I took a look at this today but it turns out to be a bit tricky due to the progression task being queued at that point in time. Currently when pausing, this task is cancelled, so some refactoring is required there if we want to change that flow.

**Reply by peppy** [reply]
Tracking via https://github.com/ppy/osu/issues/18089.
